### DISCUSSION TITLE
Static Typing Fails on Nested SQLModel

### DISCUSSION THREAD

**Discussion by martinralfreindl** [fastapi/sqlmodel#755] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
##################################### REQUIREMENTS.TXT #######################
# For running SQL queries
sqlmodel==0.0.14


############################ .VSCODE SETTINGS.JSON############################
{
    "python.defaultInterpreterPath": "${workspaceFolder}/.venv/bin/python3.11",
    "python.analysis.typeCheckingMode": "strict",
    "python.analysis.diagnosticMode": "workspace",
}


#################################### MAIN.PY #################################
from sqlmodel import SQLModel 


class User(SQLModel):
    """Gives some basic information about the user."""

    user_id: int
    
    
class Profile(object):
    """Contains social information about the user."""

    name: str


class Settings(SQLModel):
    """Contains the user's settings."""

    user: User
    profile: Profile
    use_dark_mode: bool


Settings(
    user=None, # This line does not raise static errors with pylance
    profile=None, # This line does raise static errors with pylance
    use_dark_mode=True, 
)
```

### Description

**SITUATION:** 
I recently updated to SQLModel 0.0.14 and have since started experiencing static typing issues with SQLModel. 

**COMPLICATION** 
Whenever I nest two SQLModels, static typing fails to execute correctly. 

The example above is one instance of this behavior: 

- `User` inherits from SQLModel 
- `Profile` is a regular Python class (inheriting from object)
- `Settings` contains nested User and Profile models, both of which are not optional
- When I pass None to the the user and profile properties in the `Settings` constructor, both of them should throw static errors (None is not an acceptable subtype of User and Profile after all), but for some reason the user property accepts None without issues. 
<img width="635" alt="Screenshot 2023-12-18 at 6 52 43 AM" src="https://github.com/tiangolo/sqlmodel/assets/61122332/d981e563-ca78-4450-a773-d0cf8f4bc2e7">

**QUESTION**
The code above minimally reproduces the faulty behavior on my machine. What am I doing wrong here? Is this a bug in SQLModel or my setup? 

### Operating System

macOS

### Operating System Details

I'm running all of this in VSCode

### SQLModel Version

0.0.14

### Python Version

3.11.6

**Comment by YuriiMotov** [comment]
Seems to work fine in current version
<img width="833" height="455" alt="image" src="https://github.com/user-attachments/assets/bb6b087f-3dac-4522-b42f-2b46aaf82413" />
